docarray/array/storage/base/getsetdel.py

```diff
@@ -115,6 +115,7 @@ def _del_docs_by_mask(self, mask: Sequence[bool]):
 
     def _del_all_docs(self):
         self._clear_storage()
+        self._clear_subindices()
```

Comment:
it seems to me that `subindices` should be cleared before?

Reply:
I don't think it matters. Why would it make a difference?

Reply:
semantically speaking

Reply:
or should `_clear_storage` actually call `_clear_subindices`?

Reply:
>or should _clear_storage actually call _clear_subindices?
It probably should, conceptually speaking, but `_clear_storage` can be implemented by each backend, so let's keep `_clear_subindices` outside of their responsability.
I'll switch the order though.